Adições e Correções de Componentes, Ajustes de Configuração e Melhorias no Código

Frotas opened the pull request:
## Descrição:

Este PR inclui uma série de alterações e melhorias descritas nos commits abaixo:

1. **[feat(ui/tabs): Adição de componente de navegação por `tabs`](#022af05)**
   - Implementação de um componente reutilizável para navegação com abas.
2. **[feat(ui/card): Adição do componente de `Card` reutilizável](#0169af7)**
   - Criação de um componente `Card` para melhorar a consistência do design.
3. **[build: Ajuste das configurações do NextJS de acordo com a `NODE_ENV`](#b007352)**
   - Ajustes para garantir o comportamento adequado em diferentes ambientes.
4. **[fix(page/auth): Correção de erro na análise do `localStorage`](#930853e)**
   - Tratamento para garantir que a janela seja definida antes de acessar o `localStorage`.
5. **[fix(api/auth): Correção do erro do ESLint: 'handle potential undefined'](#6386532)**
   - Ajustes nos callbacks para tratar valores indefinidos no componente `AuthCard`.
6. **[fix(lint): Correção de importação e problemas de consistência de tipo](#b152211)**
   - Ajuste de importações para cumprir regras do ESLint.
7. **[fix(ui/modal): Correção de comportamento da propriedade `defaultCloseButton`](#64db5f5)**
   - Alteração do comportamento padrão e correção do estilo aplicado.
8. **[feat(ui/badge): Adição de componente de `badge` personalizado](#0372ec0)**
   - Criação de um `badge` estilizado.
9. **[feat(ui/skeleton): Adição de um componente `Skeleton`](#49dc626)**
   - `Skeleton` loader para interfaces que carregam dados assíncronos.
10. **[fix(ui/header): Correção de redirecionamento de rota](#af4d03b)**
    - Ajuste em comportamentos de roteamento.
11. **[chore: Adição de condicional para execução do MSW](#f20e5c7)**
    - Ajustes para garantir o comportamento adequado em diferentes ambientes.
13. **[feat(utils/time): Função para formatação de data e timestamps](#eaedce6)**
    - Nova função para formatação amigável de datas e `timestamps`.

## Alterações realizadas:

- **Novos componentes**: `tabs`, `Card`, `badge`, `Skeleton`.
- **Correções gerais**: ESLint, redirecionamentos de rotas, comportamentos padrão em componentes.
- **Configurações**: Ajustes no NextJS e no MSW.
- **Melhorias utilitárias**: Função para formatação de datas.

## Testes:

- Todos os componentes foram testados manualmente e os problemas corrigidos foram verificados. 
- Recomenda-se a execução de testes adicionais para garantir a estabilidade.

## Correções

Como solicitado foram feitas as seguintes correções revertendo as alterações que trocava `type` por `interface`. Tendo em vista o padrão utilizado no projeto.

- [fix(types): Reversão de `interface` para `type` de `ProfileProps`](#a949eb6)
- [fix(types): Reversão de `interface` para `type` de `CardInfoProps`](#e74a6fa)
- [fix(types): Reversão de `interface` para `type` de `UserProps`](#8c84f40)

Frotas committed:
feat(ui/tabs): Adição de componente de navegação por `tabs`.

Frotas committed:
feat(ui/card): Adição do componente de `Card` reutilizavel

Frotas committed:
build: :wrench: Ajuste das configurações do NextJS de acordo com a `NODE_ENV`

Foi feita a adição de uma condicional para utilização de configurações diferentes para os ambientes de `Produção` e `Desenvolvimento`.

Frotas committed:
fix(page/auth): lidar com erro de análise do `localStorage` e garantir execução do lado do cliente

- Garantir que o acesso ao `localStorage` ocorra somente no lado do cliente verificando se a janela está definida.
- Prevenir erros potenciais verificando os dados do usuário analisados ​​antes de redirecionar.

Frotas committed:
fix(api/auth): Correção do erro do ESLint: `handle potential undefined callbacks in AuthCard component`

Frotas committed:
fix(lint): Correção de importação e problemas de consistência de tipo

- Ajustes de importações para cumprir com as regras ESLint.
- Garantir a consistência de tipo em toda a base de código conforme as recomendações ESLint.
- Remoção de importações não utilizadas.

Frotas committed:
fix(ui/modal): Correção de comportamento da .

- A propriedade `defaultCloseButton` foi definida como `false` por padrão.
- Alteração do comportamento padrão do `ModalContent` através do `onInteractOutside`.
- Correção de estilização

Frotas committed:
feat(ui/badge): Adição do componente de `badge` personalizavel.

Frotas committed:
feat(ui/skeleton: Adição de um componente `Skeleton`.

Frotas committed:
fix(ui/header): Correção de redirecionamento de rota.

Frotas committed:
chore: Adição de uma condicional para execução do MSW

Foi adicionado uma condicional para que o MSW seje executado somente em ambiente de desenvolvimento.

Frotas committed:
feat(utils/time): Função para formatação de data e timestamps.

A fução é responsavel por transformar `2024-01-16T22:00:00.000Z` em algo amigavel como `Jan 16, 22:00` de acordo com a linguagem definida.

0rogerinho commented on src/shared/@types/index.ts:
```diff
@@ -1,4 +1,4 @@
-export type UserProps = {
+export interface UserProps {
```
o padrão que estamos utilizando no projeto é type
```suggestion
export type UserProps = {
```

Frotas replied:
Entendi, sendo assim, teria de fazer a alteração das regras do **ESLint** para que ele mantenha e ignore isso.

0rogerinho replied:
Eu já fiz a alteração do ESLint, do dar o pull na develop

0rogerinho commented on src/shared/components/molecules/cardInfo/index.tsx:
```diff
@@ -3,7 +3,7 @@ import { cn } from "@/shared/utils/cn"
 import { variantDescription, variantTitle } from "./variants"
 
-type CardInfoProps = {
+interface CardInfoProps {
```
o padrão que estamos utilizando no projeto é type
```suggestion
export type CardInfoProps = {
```

0rogerinho commented on src/shared/components/organisms/header/components/Profile.tsx:
```diff
@@ -11,7 +11,7 @@ import {
 import { cn } from "@/shared/utils/cn"
 
-type ProfileProps = {
+interface ProfileProps {
```
o padrão que estamos utilizando no projeto é type
```suggestion
export type ProfileProps = {
```

Frotas committed:
fix(types): Reversão de `interface` para `type` de `ProfileProps`

Reverte a alteração que trocava `type` por `interface` na definição de `ProfileProps`. O padrão utilizado no projeto é o uso de `type` para definições de tipos.

Ref: Feedback da equipe para alinhar com as convenções do projeto.

Co-authored-by: devrogerinho <120338267+0rogerinho@users.noreply.github.com>

Frotas committed:
fix(types): Reversão de `interface` para `type` de `CardInfoProps `

Reverte a alteração que trocava `type` por `interface` na definição de `CardInfoProps`. O padrão utilizado no projeto é o uso de `type` para definições de tipos.

Ref: Feedback da equipe para alinhar com as convenções do projeto.

Co-authored-by: devrogerinho <120338267+0rogerinho@users.noreply.github.com>

Frotas committed:
fix(types): Reversão de `interface` para `type` de `UserProps`

Reverte a alteração que trocava `type` por `interface` na definição de `UserProps`. O padrão utilizado no projeto é o uso de `type` para definições de tipos.

Ref: Feedback da equipe para alinhar com as convenções do projeto.

Co-authored-by: devrogerinho <120338267+0rogerinho@users.noreply.github.com>
